src/m2n/M2N.hpp

```diff
@@ -147,9 +147,12 @@ class M2N {
   void createDistributedCommunication(const mesh::PtrMesh &mesh);
 
   /// Sends an array of double values from all slaves (different for each slave).
+  /// Setting the flag to true indicates that the data send is gradient data
```

Comment:
```suggestion
  /// Setting the flag "withGradient" to true indicates that the data send is gradient data
```

Reply:
Is this flag actually only needed for the name of the event? If yes, I would vote against adding it. Understandable for the test of the thesis, but not sth we might need in the long run, no?